R/learner_BART_surv_bart.R

```diff
@@ -0,0 +1,212 @@
+          b = p_dbl(default = 1L, tags = "train"),
+          augment = p_lgl(default = FALSE, tags = "train"),
+          rho = p_dbl(default = NULL, special_vals = list(NULL), tags = "train"),
+          usequants = p_lgl(default = FALSE, tags = "train"),
+          rm.const = p_lgl(default = TRUE, tags = "train"),
+          type = p_fct(levels = c("pbart", "lbart"), default = "pbart", tags = "train"),
+          ntype = p_int(lower = 1, upper = 3, tags = "train"),
+          k = p_dbl(default = 2.0, lower = 0, tags = "train"),
+          power = p_dbl(default = 2.0, lower = 0, tags = "train"),
+          base = p_dbl(default = 0.95, lower = 0, upper = 1, tags = "train"),
+          offset = p_dbl(default = NULL, special_vals = list(NULL), tags = "train"),
+          ntree = p_int(default = 50L, lower = 1L, tags = "train"),
+          numcut = p_int(default = 100L, lower = 1L, tags = "train"),
+          ndpost = p_int(default = 1000L, lower = 1L, tags = "train"),
+          nskip = p_int(default = 250L, lower = 0L, tags = "train"),
+          keepevery = p_int(default = 10L, lower = 1L, tags = "train"),
+          printevery = p_int(default = 100L, lower = 1L, tags = "train"),
+          seed = p_int(default = 99L, tags = "train"),
+          mc.cores = p_int(default = 2L, lower = 1L, tags = c("train", "predict")),
+          nice = p_int(default = 19L, lower = 0L, upper = 19L, tags = c("train", "predict")),
+          openmp = p_lgl(default = TRUE, tags = "predict"),
+          quiet = p_lgl(default = TRUE, tags = "predict"),
+          importance = p_fct(default = "count", levels = c("count", "prob"), tags = "train"),
+          which.curve = p_dbl(special_vals = list("mean"), tags = "predict")
+        )
+
+        # custom defaults
+        param_set$values = list(mc.cores = 1, quiet = TRUE, importance = "count",
+          which.curve = 0.5) # 0.5 quantile => median posterior
+
+        super$initialize(
+          id = "surv.bart",
+          packages = "BART",
+          feature_types = c("logical", "integer", "numeric"),
+          predict_types = c("crank", "distr"),
+          param_set = param_set,
+          properties = c("importance", "missings"),
+          man = "mlr3extralearners::mlr_learners_surv.bart",
+          label = "Bayesian Additive Regression Trees"
+        )
+      },
+
+      #' @description
+      #' Two types of importance scores are supported based on the initial value
+      #' for the parameter `importance`:
+      #' 1. `prob`: The mean selection probability of each feature in the trees,
+      #' extracted from the slot `varprob.mean`.
+      #' If `sparse = FALSE` (default), this is a fixed constant.
+      #' Recommended to use this option when `sparse = TRUE`.
+      #' 2. `count`: The observed count of each feature in the trees, extracted
+      #' from the slot `varcount.mean`.
+      #' This is the default importance scores.
```

Comment:
Is it really the observed count? (I am asking because the variable is called `varcount.mean`.)

Reply:
The `mean`, I also rephrased it, thanks!